app/controllers/donations_controller.rb

```diff
@@ -8,6 +8,17 @@ def index
 		render json: Donation.all
 	end
 
+    def donations_history
+        id = params[:donor_id].to_i
+        closed_donations_in_db = Donation.where(:status => [DonationStatus::CLOSED, DonationStatus::EXPIRED], donor_id: id).order("updated_at DESC").limit(50)
+        render json: closed_donations_in_db
```

Comment:
In both of these methods, I think we should probably prevent users for accessing anything other than their own data.
```suggestion
     id = params[:donor_id].to_i
				authorized_id = decoded_token[0]['donor_id']
				if authorized_id != id
					return render json: { error: 'unauthorized'}, status: :unauthorized
				end
        closed_donations_in_db = Donation.where(:status => [DonationStatus::CLOSED, DonationStatus::EXPIRED], donor_id: id).order("updated_at DESC").limit(50)
        render json: closed_donations_in_db
```

Reply:
> Great job! I made a few small suggestions where I think we can make the code better, but everything already works as expected and nice job with the tests.

Thank you!!